website/www/site/content/en/documentation/sdks/python-pipeline-dependencies.md

```diff
@@ -95,43 +95,47 @@ If your pipeline uses packages that are not available publicly (e.g. packages th
 
 Often, your pipeline code spans multiple files. To run your project remotely, you must group these files as a Python package and specify the package when you run your pipeline. When the remote workers start, they will install your package. To group your files as a Python package and make it available remotely, perform the following steps:
 
-1. Create a [setup.py](https://pythonhosted.org/an_example_pypi_project/setuptools.html) file for your project. The following is a very basic `setup.py` file.
+1. Create a [pyproject.toml](https://packaging.python.org/en/latest/tutorials/packaging-projects/) file for your project. The following is a very basic `pyproject.toml` file.
 
-        import setuptools
+        [project]
+        name = "PACKAGE-NAME"
+        version = "PACKAGE-VERSION"
+        dependencies = [
+          # List Python packages your pipeline depends on.
+        ]
+
+2. If needed by your environment, create a setup.py file for your project.
 
-        setuptools.setup(
-           name='PACKAGE-NAME',
-           version='PACKAGE-VERSION',
-           install_requires=[
-             # List Python packages your pipeline depends on.
-           ],
-           packages=setuptools.find_packages(),
-        )
+        # Note that the package is completely defined by pyproject.toml.
+        # This file is optional.
+        import setuptools
+        setuptools.setup()
 
-2. Structure your project so that the root directory contains the `setup.py` file, the main workflow file, and a directory with the rest of the files, for example:
+3. Structure your project so that the root directory contains the `pyproject.toml`, the `setup.py` file, and a `src/` directory with the rest of the files. For example:
 
         root_dir/
+          pyproject.toml
           setup.py
-          main.py
-          my_package/
-            my_pipeline_launcher.py
-            my_custom_dofns_and_transforms.py
-            other_utils_and_helpers.py
+          src/
+            main.py
+            my_package/
+              my_pipeline_launcher.py
+              my_custom_dofns_and_transforms.py
+              other_utils_and_helpers.py
 
     See [Juliaset](https://github.com/apache/beam/tree/master/sdks/python/apache_beam/examples/complete/juliaset) for an example that follows this project structure.
 
-3. Install your package in the submission environment, for example by using the following command:
+4. Install your package in the submission environment, for example by using the following command:
 
         pip install -e .
 
-4. Run your pipeline with the following command-line option:
+5. Run your pipeline with the following command-line option:
```

Comment:
If you use a custom container (link to {#custom-containers} ), copy and install the package in the container as well. Otherwise, run your pipeline with the following command-line option: